### PR TITLE
Add basic documentation for multi node training

### PR DESCRIPTION
**Pull request by SumanthRH**
# What does this PR do?

Adds some basic documentation for multi-node training with 🤗 Accelerate. I've also added a small comment on running code on CPU, which I think is helpful but is not talked about in the tutorials. 

cc @muellerzr

### CODE REVIEW

**Review by pacman100 — APPROVED** [review]
Thank you @SumanthRH for improving the docs wrt Multinode training and CPU training 🤗

File: docs/source/basic_tutorials/launch.md [hunk 1]
```diff
@@ -200,3 +209,18 @@ Launching a script from the location of that custom yaml file looks like the fol
...
 ```
+
+## Multi-node training
+Multi-node training with 🤗Accelerate is similar to [multi-node training with torchrun](https://pytorch.org/tutorials/intermediate/ddp_series_multinode.html). The simplest way to launch a multi-node training run is to do the following:
+
+- Copy your codebase and data to all nodes. (or place them on a shared filesystem)
+- Setup your python packages on all nodes.
+- Run `accelerate config` on all nodes. Here, after specifying the number of nodes, you will be asked to specify the rank of each node (this is 0 for the main/master node, and 1,2,3,etc for worker nodes), along with the IP address and port for the main process. This is required for the worker nodes to communicate with the main process. 
+
+Once you have done this, you can start your multi-node training run by running `accelerate launch` on all nodes.
```
Comment (muellerzr) [inline docs/source/basic_tutorials/launch.md:220]:
```suggestion
Once you have done this, you can start your multi-node training run by running `accelerate launch` (or `torchrun`) on all nodes.

<Tip>
    It is required that the command be ran on all nodes for everything to start, not just running it from the main node. You can use something like SLURM or a different process executor to wrap around this requirement and call everything from a single command.
</Tip>
```

**Review by muellerzr — APPROVED** [review]
Thanks for this wonderful addition!

I left a few suggestions, after merging we are good to go :)

**Comment by SumanthRH** [comment]
@muellerzr suggestions merged! I also added a link to the new blog post on [multi-node training with FSDP](https://huggingface.co/blog/ram-efficient-pytorch-fsdp).

**Comment by muellerzr** [comment]
Thanks for the PR! 🤗

**Pull request merged**